src/engine/source/indexerconnector/test/component/indexerConnector_test.cpp

```diff
@@ -200,6 +203,28 @@ TEST_F(IndexerConnectorTest, ConnectionWithSslCredentials)
     auto indexerConnector {IndexerConnector(indexerConfig)};
 }
 
+/**
+ * @brief Test the connection to an available server with SSL credentials.
+ *
+ * @note The SSL data is a dummy one and there are no functionality checks here. The target of this test is to increase
+ * the test coverage.
+ *
+ */
+TEST_F(IndexerConnectorTest, ConnectionWithCertsArray)
+{
+    IndexerConnectorOptions indexerConfig {
+        .name = INDEXER_NAME,
+        .hosts = {A_ADDRESS},
+        .sslOptions = {.cacert = {"/etc/filebeat/certs/root-ca.pem", "/etc/filebeat/certs/root-ca-two.pem"},
+                       .cert = "/etc/filebeat/certs/filebeat.pem",
+                       .key = "/etc/filebeat/certs/filebeat-key.pem"},
+        .timeout = INDEXER_TIMEOUT};
+
+    // Create connector and wait until the connection is established.
+    // Throw is expected if the certs are not found.
```

Comment:
I think a better option is to generate some dummy pem files and see if they were merged into one

Reply:
I couldn't make it due to this check

https://github.com/wazuh/wazuh/blob/c1befeb4fc1399a8bd9552ea2a1daad83f18f012/src/engine/source/indexerconnector/src/indexerConnector.cpp#L82-L88

We could remove it or add the user and group on the test step, what do you think?

Reply:
Yes I think we are okay with that